samza-kafka/src/main/scala/org/apache/samza/checkpoint/kafka/KafkaCheckpointManager.scala

```diff
@@ -187,92 +174,71 @@ class KafkaCheckpointManager(
     checkpoints.toMap /* of the immutable kind */
   }
 
+  private def getEarliestOffset(topic: String, partition: Partition): String = {
+    val systemAdmin = getSystemAdmin()
+    val metaDataMap: java.util.Map[String, SystemStreamMetadata] = systemAdmin.getSystemStreamMetadata(Collections.singleton(topic))
+    val checkpointMetadata: SystemStreamMetadata = metaDataMap.get(topic)
+    if(checkpointMetadata == null)
+      throw new SamzaException("Cannot get metadata for system=%s, topic=%s" format (systemName, topic))
+
+    val partitionMetaData = checkpointMetadata.getSystemStreamPartitionMetadata().get(partition);
+    if(partitionMetaData == null)
+      throw new SamzaException("Cannot get partitionMetaData for system=%s, topic=%s" format (systemName, topic))
+
+    partitionMetaData.getOldestOffset
+  }
 
   /**
-   * Common code for reading both changelog partition mapping and change log
+   * Code for reading checkpoint log
    *
    * @param entryType What type of entry to look for within the log key's
    * @param handleEntry Code to handle an entry in the log once it's found
    */
   private def readLog(entryType:String, shouldHandleEntry: (KafkaCheckpointLogKey) => Boolean,
                       handleEntry: (ByteBuffer, KafkaCheckpointLogKey) => Unit): Unit = {
-    retryBackoff.run[Unit](
```

Comment:
why do you remove retryBackoff?

Reply:
It used to retry direct requests to the Kafka brokers, but systemConsumer makes poll() requests to the BlockingEnvelopeMap, and, since it is local only call, there is no point in retries.
Same if we go to a high level consumer in Kafka. It will take care of retries.